packages/core/data-transfer/src/engine/index.ts

```diff
@@ -316,8 +316,34 @@ class TransferEngine<
     aggregate?: {
       size?(value: unknown): number;
       key?(value: unknown): string;
-    }
+    },
+    perChunk?: boolean
   ) {
+    if (perChunk) {
```

Comment:
I'm not a big fan of always adding more options and branches to the flow, could we try to simplify it a bit?

For instance, since the perChunk option seems to only be useful if coupled with the asset stage, could we remove the option and instead branch based on the stage value?

Reply:
I'd rather have the stage itself decide, since it needs to provide additional info anyway. And even though it's only on assets stage for now, I thought it would be better to keep it generic so that we could use it anywhere else in the future, no reason for it to be tied to assets necessarily.